From 6c9e43fd8c7e49d51d9237ce488bf68c5d455abc Mon Sep 17 00:00:00 2001
From: Tobias Klinke <TobiasAlexander.Klinke@baslerweb.com>
Date: Fri, 12 Feb 2021 12:08:41 +0100
Subject: [PATCH 20/24] media: mxc: change mx6s_fmt.bpp to bits per pixel

The previous meaning was bytes per pixel.
This commit prepares for adding YUV420 support
which will be 12 bits per pixel.
---
 .../media/platform/mxc/capture/mx6s_capture.c  | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/media/platform/mxc/capture/mx6s_capture.c b/drivers/media/platform/mxc/capture/mx6s_capture.c
index 19ab7055f..f50b1bcb6 100644
--- a/drivers/media/platform/mxc/capture/mx6s_capture.c
+++ b/drivers/media/platform/mxc/capture/mx6s_capture.c
@@ -275,43 +275,43 @@ static struct mx6s_fmt formats[] = {
 		.fourcc		= V4L2_PIX_FMT_UYVY,
 		.pixelformat	= V4L2_PIX_FMT_UYVY,
 		.mbus_code	= MEDIA_BUS_FMT_UYVY8_2X8,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "YUYV-16",
 		.fourcc		= V4L2_PIX_FMT_YUYV,
 		.pixelformat	= V4L2_PIX_FMT_YUYV,
 		.mbus_code	= MEDIA_BUS_FMT_YUYV8_2X8,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "YUV32 (X-Y-U-V)",
 		.fourcc		= V4L2_PIX_FMT_YUV32,
 		.pixelformat	= V4L2_PIX_FMT_YUV32,
 		.mbus_code	= MEDIA_BUS_FMT_AYUV8_1X32,
-		.bpp		= 4,
+		.bpp		= 32,
 	}, {
 		.name		= "RAWRGB8 (SBGGR8)",
 		.fourcc		= V4L2_PIX_FMT_SBGGR8,
 		.pixelformat	= V4L2_PIX_FMT_SBGGR8,
 		.mbus_code	= MEDIA_BUS_FMT_SBGGR8_1X8,
-		.bpp		= 1,
+		.bpp		= 8,
 	}, {
 		.name		= "RAWRGB8 (SRGGB8)",
 		.fourcc		= V4L2_PIX_FMT_SRGGB8,
 		.pixelformat	= V4L2_PIX_FMT_SRGGB8,
 		.mbus_code	= MEDIA_BUS_FMT_SRGGB8_1X8,
-		.bpp		= 1,
+		.bpp		= 8,
 	}, {
 		.name		= "Grey 10bit",
 		.fourcc		= V4L2_PIX_FMT_Y10,
 		.pixelformat = V4L2_PIX_FMT_Y10,
 		.mbus_code	= MEDIA_BUS_FMT_Y10_1X10,
-		.bpp		= 2,
+		.bpp		= 16,
 	}, {
 		.name		= "RAW10 (SBGGR10)",
 		.fourcc		= V4L2_PIX_FMT_SBGGR10,
 		.pixelformat = V4L2_PIX_FMT_SBGGR10,
 		.mbus_code	= MEDIA_BUS_FMT_Y10_1X10,
-		.bpp		= 2,
+		.bpp		= 16,
 	}
 };
 
@@ -1544,8 +1544,8 @@ static int mx6s_vidioc_try_fmt_vid_cap(struct file *file, void *priv,
 	if (pix->field != V4L2_FIELD_INTERLACED)
 		pix->field = V4L2_FIELD_NONE;
 
-	pix->sizeimage = fmt->bpp * pix->height * pix->width;
-	pix->bytesperline = fmt->bpp * pix->width;
+	pix->bytesperline = DIV_ROUND_UP(fmt->bpp * pix->width, 8);
+	pix->sizeimage = pix->bytesperline * pix->height;
 
 	return ret;
 }
-- 
2.31.1

